frontend/src/pages/NewRunV2.test.tsx

```diff
@@ -500,12 +465,13 @@ describe('NewRunV2', () => {
       const choosePipelineButton = screen.getAllByText('Choose')[0];
       fireEvent.click(choosePipelineButton);
 
-      const getPipelineSpy = jest.spyOn(Apis.pipelineServiceApi, 'getPipeline');
-      getPipelineSpy.mockImplementation(() => NEW_TEST_PIPELINE);
-
       const expectedPipeline = await screen.findByText(NEW_TEST_PIPELINE_NAME);
       fireEvent.click(expectedPipeline);
 
+      await waitFor(() => {
+        expect(getPipelineSpy).toHaveBeenCalled();
+      });
```

Comment:
For my education, why do we need to add this code now?

Reply:
Simple answer: this is related to ResourceSelectorV2. 

In ResourceSelector(v1), the component will return the selected resource (base resource) back to its parent component (which is PipelineDialog(v1) here). Because `ApiPipeline` has same field name (`id`, `name` etc.), the returned base resource from ResourceSelector can be used directly in PipelineDialog(v1) (note: PipelineDialog(v1) will return the `ApiPipeline` to its parent component NewRun(v1) afterward.

As for V2beta1Pipeline, it has different name field as base resource, so we discuss to create a **converter** which works well when convert `apiListResponse` to base resource. However, it will miss some information when convert back from base resource to API object (pipeline, version, experiment). There are two solutions for this bug: 
1. Add all missing info as optional field in base resource interface in ResourceSelectorV2, then convert it back to API object.
2. (Currently using) return `selectedId` rather than `selected base resource` back to its parent component (PipelineDialogV2). Once the pipeline dialog get the returned selectId from resource selector, it then call getPipeline() API again to retrieve the selectedPipeline which is returned back to parent component (NewRunV2). Also, other selectors like experiment and version in NewRunV2 are using the same logic.